src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

```diff
@@ -16,617 +15,590 @@
+                            break;
+                        case EVENT_START_FIGHT:
+                            CombatStart();
+                            break;
+                        case EVENT_RELEASED:
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_NOT_SELECTABLE);
+                            me->SetInCombatWithZone();
+                            instance->SetData(DATA_MANTICRON_CUBE, ACTION_ENABLE);
+                            events.ScheduleEvent(EVENT_CLEAVE, Seconds(10));
+                            events.ScheduleEvent(EVENT_BLAST_NOVA, Seconds(60));
+                            events.ScheduleEvent(EVENT_BLAZE, Seconds(20));
+                            events.ScheduleEvent(EVENT_QUAKE, Seconds(35));
+                            events.ScheduleEvent(EVENT_BERSERK, Minutes(20));
+                            break;
+                        case EVENT_COLLAPSE:
+                            instance->SetData(DATA_COLLAPSE_2, ACTION_ENABLE);
+                            events.ScheduleEvent(EVENT_DEDRIS_KNOCKDOWN, Seconds(4));
+                            break;
+                        case EVENT_DEDRIS_KNOCKDOWN:
+                            if (Creature* trigger = instance->GetCreature(DATA_WORLD_TRIGGER))
+                            {
+                                trigger->CastSpell(trigger, SPELL_DEBRIS_KNOCKDOWN, true);
+                                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                                me->SetReactState(REACT_AGGRESSIVE);
+                                events.ScheduleEvent(EVENT_DEDRIS, Seconds(20));
+                            }
+                            break;
+                        case EVENT_DEDRIS:
+                            DoCast(SPELL_DEDRIS_SERVERSIDE);
+                            events.Repeat(Seconds(20));
+                            break;
+                        case EVENT_NEARLY_EMOTE:
+                            Talk(EMOTE_NEARLY_FREE, me);
+                            break;
+                        case EVENT_BLAST_NOVA:
+                            Talk(EMOTE_BLAST_NOVA, me);
+                            DoCast(SPELL_BLAST_NOVA);
+                            events.Repeat(Seconds(55));
+                            break;
+                        case EVENT_TAUNT:
+                            Talk(SAY_TAUNT);
+                            events.Repeat(Minutes(4), Minutes(5));
+                            break;
+                        default:
+                            break;
                     }
-                    return;
-                }
 
-                if (Despawn_Timer <= diff)
-                {
-                    me->DespawnOrUnsummon();
+                    if (me->HasUnitState(UNIT_STATE_CASTING))
+                        return;
                 }
-                else Despawn_Timer -= diff;
 
                 if (!UpdateVictim())
```

Comment:
He execute events before fight, so if i put UpdateVictim() before events, some events will not execute